index.js

```diff
@@ -23,42 +23,62 @@ module.exports = {
 		this._app = appOrAddon.app || appOrAddon;
 		this.options = new Config(this._app, this.ui);
 
-		this.importPolyfills();
+		this.importPolyfill();
 	},
 
-	importPolyfills() {
-		const { polyfillBundle, buildForProduction } = this.options;
+	importPolyfill() {
+		const { polyfillBundle } = this.options;
 
-		const webcomponentsjsPath = path.join(this._app.bowerDirectory, 'webcomponentsjs');
-		const customElementsEs5Adapter = path.join(webcomponentsjsPath, 'custom-elements-es5-adapter.js');
-		const webcomponentsjsPolyfill = path.join(webcomponentsjsPath, `webcomponents-${polyfillBundle}.js`);
+		if (polyfillBundle && polyfillBundle !== 'none') {
+			const webcomponentsjsPath = path.join(this._app.bowerDirectory, 'webcomponentsjs');
+			const webcomponentsjsPolyfill = path.join(webcomponentsjsPath, `webcomponents-${polyfillBundle}.js`);
 
-		// Import custom-elements-es5-adapter if ES6 Classes are transpiled to ES5
-		// for browsers that natively support Custom Elements
-		// https://github.com/webcomponents/webcomponentsjs#custom-elements-es5-adapterjs
-		if (buildForProduction.enabled && buildForProduction.build.js.compile) {
-			this._app.import(customElementsEs5Adapter, { options: 'prepend' });
+			this._app.import(webcomponentsjsPolyfill, { options: 'prepend' });
 		}
-
-		this._app.import(webcomponentsjsPolyfill, { options: 'prepend' });
 	},
 
 	// insert polyfills and bundled elements
 	contentFor(type, config) {
-		if (type !== 'head') {
-			return null;
-		}
+		if (type === 'head-footer') {
+			const headContents = [];
+			const { globalPolymerSettings, buildForProduction } = this.options;
 
-		const headContents = [];
-		const { globalPolymerSettings } = this.options;
+			if (globalPolymerSettings) {
+				headContents.push(`<script>window.Polymer = ${JSON.stringify(globalPolymerSettings)};</script>`);
+			}
 
-		if (globalPolymerSettings) {
-			headContents.push(`<script>window.Polymer = ${JSON.stringify(globalPolymerSettings)};</script>`);
+			if (buildForProduction.enabled && buildForProduction.build.js.compile) {
```

Comment:
The polyfill should be always injected into the application. There shouldn't been differences between development and production.

Reply:
webcomponentsjs polyfill (webcomponents-lite) is always injected in vendors but custom-elements-es5-adapter is injected only if `buildForProduction` (is an addon option) is enabled. A user may not want to transpile the bundle depending on the application targets (maybe is not needed), so the bundle should not be always transpiled / minified.

Reply:
I don't agree, you may want not to transpile but your criterum should be other than `buildForProduction`. Anyway, we can revisit this discussion later

Reply:
`buildForProduction.build.js.compile` The compile option can be `false`.